manifests/install.pp

```diff
@@ -4,18 +4,24 @@
 class qpid::install {
   include qpid::tools
 
+  if $qpid::ensure == 'absent' {
+    $_package_ensure = $qpid::ensure
```

Comment:
```suggestion
    $_package_ensure = 'purged'
```

It turns out the yum provider uses `rpm -e` when `ensure => absent`, which explains the behavior I saw in the acceptance test and why my first idea did not resolve it. 

However `ensure => purged` resolved the problem by actually using yum to remove the packages. Afterwards on the test system I verified that the dependencies were correctly removed:

```
[root@centos7-64-1 ~]# yum history info 9
Loaded plugins: fastestmirror
Transaction ID : 9
Begin time     : Fri Apr  2 04:29:43 2021
Begin rpmdb    : 343:da3c175c120c70db164a3ee5d245a119f472f947
End time       :            04:29:44 2021 (1 seconds)
End rpmdb      : 339:2d5a594ccf2ee058ba67b46b501e4bdc44fab71a
User           : root <root>
Return-Code    : Success
Command Line   : -y erase qpid-cpp-client
Transaction performed with:
    Installed     rpm-4.11.3-43.el7.x86_64                      @anaconda
    Installed     yum-3.4.3-167.el7.centos.noarch               @anaconda
    Installed     yum-plugin-fastestmirror-1.1.31-53.el7.noarch @anaconda
Packages Altered:
    Erase python2-qpid-qmf-1.39.0-1.el7.x86_64 @epel
    Erase qpid-cpp-client-1.39.0-1.el7.x86_64  @epel
    Erase qpid-qmf-1.39.0-1.el7.x86_64         @epel
    Erase qpid-tools-1.39.0-1.el7.noarch       @epel
history info
[root@centos7-64-1 ~]# yum history info 8
Loaded plugins: fastestmirror
Transaction ID : 8
Begin time     : Fri Apr  2 04:29:42 2021
Begin rpmdb    : 345:b800c4ff8d099fe7d7b6f1b9715103f483b07f0d
End time       :            04:29:43 2021 (1 seconds)
End rpmdb      : 343:da3c175c120c70db164a3ee5d245a119f472f947
User           : root <root>
Return-Code    : Success
Command Line   : -y erase qpid-cpp-server
Transaction performed with:
    Installed     rpm-4.11.3-43.el7.x86_64                      @anaconda
    Installed     yum-3.4.3-167.el7.centos.noarch               @anaconda
    Installed     yum-plugin-fastestmirror-1.1.31-53.el7.noarch @anaconda
Packages Altered:
    Erase qpid-cpp-server-1.39.0-1.el7.x86_64             @epel
    Erase qpid-cpp-server-linearstore-1.39.0-1.el7.x86_64 @epel
Scriptlet output:
   1 warning: /etc/qpid/qpidd.conf saved as /etc/qpid/qpidd.conf.rpmsave
history info
```